### ISSUE TITLE
Realm.io conflict with AnyObject subscript

### ISSUE THREAD

**Issue by happiehappie** [realm/realm-cocoa#2623]
I'm about to implement Realm models in my Swift 2.0 app. However, once I pod install Realm. My AnyObject subscripts for my app will cause a compile error

"Cannot subscript a value of type 'AnyObject' with an index of type 'String'" 
"Cannot subscript a value of type 'RLMProperty' with an index of type 'String'"

let outlet: AnyObject = (self.campaigns[indexPath.section]["surveys"]!![indexPath.row])["outlet"]!!

    // Configure the cell...
    campaignCell.outletID.text = outlet["code"] as? String
    campaignCell.outletNameLabel.text = outlet["name"] as? String
    //campaignCell.outletAddressLabel.text = outlet["outlet"]!!["address"] as? String
    campaignCell.outletStatusLabel.text = (self.campaigns[indexPath.section]["surveys"]!![indexPath.row])["progress"]!!["name"] as? String

    campaignCell.outletStatusLabel.textColor = UIColor.colorWithHexString(((self.campaigns[indexPath.section]["surveys"]!![indexPath.row])["progress"]!!["color"] as? String)!)

**Comment by mrackwitz** [comment]
>My AnyObject subscripts for my app will cause a compile error

There are no subscripts with Swift by default for `AnyObject`. So you would need to provide some further info, how you have made them possible in the first place.
It looks to me like you have defined a custom postfix operator `!!`, which force-unwraps the optional `AnyObject` argument and force-casts it to another subscript-able type `[String : AnyObject]`. I don't see how that should directly conflict with any definition given by Realm. Are any of the involved objects `RLMObject` instances? Do you access the object schema anywhere?

**Comment by bigfish24** [comment]
@happiehappie I noticed you posted this on [StackOverflow](http://stackoverflow.com/questions/32899683/realm-io-with-anyobject-subscript). We would prefer to finish this conversation over there so that others can more easily benefit from the answer. As a result, I am going to close this issue for now.

**Comment by Eldstorm** [comment]
I don't have enough reputation on StackOverflow so I can't post there. However the same thing happened to my too now. Everything worked fine when I used “import RealmSwift” then I “Cleaned Build Folder” and after that the issue acquired. 

This is one line:
let title = obj[“title”] as? String 

And it gives me error:
Cannot subscript a value of type 'AnyObject' with an index of type 'String'

I did not used pod to install Realm, I did it manually. 

I get around this issue if I comment out import RealmSwift and then “clean my project”,
after that I uncomment it again and then build. It works again but every time I “Clean Build Folder” same error happens.
